### PR TITLE
Support for hyperparameters with JSON serialized and non serialized keys altogether

### PR DESCRIPTION
**Pull request by mvsusp**
## Description of changes:
- Support for hyperparameters with JSON serialized and non serialized keys altogether
- Add new SageMaker hyperparameter **_tuning_objective_metric**
- Add new env variable **TRAINING_JOB_NAME**

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: test/unit/test_environment.py [hunk 1]
```diff
@@ -49,7 +49,8 @@
     'sagemaker_program':                   'main.py',
     'sagemaker_submit_directory':          'imagenet',
     'sagemaker_enable_cloudwatch_metrics': True,
-    'sagemaker_container_log_level':       logging.WARNING
+    'sagemaker_container_log_level':       logging.WARNING,
+    '_tuning_objective_metric':            'loss:3.4'
```
Comment (laurenyu) [inline test/unit/test_environment.py]: nit: if you add a comma at the end of the line here, it'll make a cleaner diff the next time you need to add something to the dict

File: test/unit/test_environment.py [hunk 2]
```diff
@@ -68,6 +69,17 @@ def test_read_key_serialized_hyperparameters():
     assert _env.read_hyperparameters() == ALL_HYPERPARAMETERS
 
 
+def test_read_key_serialized_and_non_key_serialized_hyperparameters():
+    hps_to_serialize = ('default_user_module_name', 'sagemaker_job_name', 'sagemaker-training-job',
```
Comment (laurenyu) [inline test/unit/test_environment.py]: is `'sagemaker-training-job'` serving any purpose here?

Comment (mvsusp) [reply test/unit/test_environment.py]: The test asserts that parsing the hyperparameters file work when some hyperparameters are JSON serialized and some are not.

So, these keys could any keys in the list of hyperparameters.

File: test/unit/test_environment.py (hunk 1 above)
Comment (mvsusp) [reply test/unit/test_environment.py]: Got it. I am working in a pylint file. I will add the the comma requirement when I enable linting.

File: test/unit/test_environment.py (hunk 2 above)
Comment (laurenyu) [reply test/unit/test_environment.py]: but `'sagemaker-training-job'` is just a value in `SAGEMAKER_HYPERPARAMETERS` - I don't see it being used as a key anywhere

Comment (mvsusp) [reply test/unit/test_environment.py]: ```ALL_HYPERPARAMETERS == SAGEMAKER_HYPERPARAMETERS + USER_HYPERPARAMETERS```:

https://github.com/mvsusp/sagemaker-containers/blob/6e5ca1539d49e48888094b33ad4f085f66a6423b/test/unit/test_environment.py#L56

The test can less complicated. I changed the testes to make the test setup clear for better readability.

**Pull request merged**